Modules/Filtering/Smoothing/test/itkDiscreteGaussianImageFilterTest2.cxx

```diff
@@ -53,30 +61,33 @@ itkDiscreteGaussianImageFilterTestA(const char * inputFilename, float sigma, con
 int
 itkDiscreteGaussianImageFilterTest2(int argc, char * argv[])
 {
-  if (argc != 6)
+  if (argc < 5)
   {
     std::cerr << "Missing parameters." << std::endl;
     std::cerr << "Usage:" << std::endl;
-    std::cerr << itkNameOfTestExecutableMacro(argv)
-              << " imageDimension vectorDimension inputFilename sigma outputFilename" << std::endl;
+    std::cerr << itkNameOfTestExecutableMacro(argv) << " imageDimension vectorDimension inputFilename outputFilename"
+              << " [sigma] [kernelError] [kernelWidth] " << std::endl;
     return EXIT_FAILURE;
   }
 
   unsigned int img_dim = std::stoi(argv[1]);
-  unsigned int vec_dim = std::stoi(argv[2]);
-
   if (img_dim != 2)
   {
     std::cerr << "This test only supports 2D image for demo! exiting ..." << std::endl;
     return EXIT_FAILURE;
   }
 
+  unsigned int vec_dim = std::stoi(argv[2]);
   if (vec_dim != 1 && vec_dim != 3)
   {
     std::cerr << "This test only supports 3-channel image or 1-channel image for demo! Exiting ... " << std::endl;
     return EXIT_FAILURE;
   }
 
+  float        sigma = (argc >= 6) ? std::stof(argv[5]) : 0.0;
```

Comment:
I find it easier to spot inconsistencies if the same number appears in the condition and the index, e.g. `argc > 5 ... argv[5]`.

Reply:
Fair enough, will fix.